src/status_im/contexts/wallet/wallet_connect/events/session_responses.cljs

```diff
@@ -94,13 +95,24 @@
              :tx-hash    tx-hash
              :tx-params  tx-args
              :on-error   #(rf/dispatch [:wallet-connect/on-sign-error %])
-             :on-success #(rf/dispatch [:wallet-connect/finish-session-request %])}]]})))
+             :on-success #(rf/dispatch [:wallet-connect/respond-sign-message-success %])}]]})))
+
+(rf/reg-event-fx :wallet-connect/respond-sign-message-success
+ (fn [{:keys [db]} [result]]
+   (let [network (data-store/get-network-from-request db)]
+     {:fx [[:dispatch [:wallet-connect/finish-session-request result]]
+           [:dispatch
+            [:centralized-metrics/track :metric/dapp-sign
+             {:network network
+              :action  :approved
+              :result  :success}]]]})))
```

Comment:
I believe it would be better to categorize them by main topics (sign, send, transfer), and also include the specific `method` for cases like ‘sign,’ which can vary. IMO using a single event for all categories might not be the best approach.

Reply:
for point 1 i'm sending event according to their category and result, so sign-success and sign-error, also send-success and send-error, i will be messy if i want to only use `:wallet-connect/finish-session-request` for handling them

Reply:
Hey @clauxx, We have an acceptance area, and I’ve implemented it to the best of my ability to meet those criteria. I understand that everyone has their own preferences when it comes to data visualization and presentation. However, could you please point out any specific issues with this implementation regarding the acceptance criteria, code readability, or performance?

For example, it seems clear that eth_signTransaction falls under the ‘sign’ category, while eth_sendTransaction belongs to ‘send’.

It might be more efficient for us to first align on the preferred data structure, and then I can adjust the implementation accordingly to avoid wasting time. I don’t have a strong preference regarding the structure; I just went with what made the most sense to me. 
I just missed buy eth button event and will add it. 
https://www.notion.so/Event-Tracking-Specification-Mobile-11a8f96fb65c8032a05efcebea3a1ce6”

Reply:
@mohsen-ghafouri 
> We have an acceptance area, and I’ve implemented it to the best of my ability to meet those criteria. I understand that everyone has their own preferences when it comes to data visualization and presentation. However, could you please point out any specific issues with this implementation regarding the acceptance criteria, code readability, or performance?

I don't feel that Cristian's question was about the correctness. His argument is strong in saying that we can get the information you are collecting (categorization of request types) using the `finish-session-request` handler. 

Instead of event names like: `eth_sign`, `eth_send`, we'll have and event name like `finish-session-request {:method :eth_sign}`. His solution is slightly more elegant. Although your solution matches the acceptance criteria too, I think Cristian's feedback has merit.


@clauxx 
>  think we should improve the event-id names :metric/dapp-sign and :metric/dapp-send, cause it's a bit weird that we're differentiating them with "sign" and "send" (likely inferred from the method name), while in all cases a response will be sent back to the dapp. Would be better to unify them into one event and call it smth like :metric/send-dapps-response

`dapp-sign` and `dapp-send` feel a little incomplete to me too. I have checked the code, and I saw we are emitting these events manually. I think we should avoid it as much as we can. Hooking into existing events is a more clear solution. What do you think @mohsen-ghafouri ?

@mohsen-ghafouri 
> I believe it would be better to categorize them by main topics (sign, send, transfer), and also include the specific method for cases like ‘sign,’ which can vary. IMO using a single event for all categories might not be the best approach.

Why do you think that using a single event is not the best approach? We have an event name and inside it's body, we'll have the type of event ? 

--

This seems to be a long discussion, if you want, we can do it synchronously on call ?

Reply:
Hey, coming a bit late to this discussion, I have few points to share. Overall, I think we should take the opportunity at this point of development with metrics to do the right thing and not be too ad-hoc.

It's natural to have disagreements about how to model events and event metadata. This is in part similar to how we can structure a database, where many constraints should be taken in consideration and there are multiple solutions to the same problem.

- It is advisable for us to create a tracking plan https://docs.mixpanel.com/docs/tracking-best-practices/tracking-plan#overview. I see already that in this PR and possibly in future ones, we will disagree on how the events should be modeled, therefore it's important that, before development starts, we agree on the data model and the business questions we want to answer. And include the users' who will analyse the data (such as Volo).

- At this stage of the product I do think we should collect only the bare minimum of data because we only need to know roughly what users are doing in the app. My perception is that we are tracking too much already in too much detail. Only once we identify hotspots we should consider tracking in more detail to answer more specific questions. We could simplify this PR.

- Prefer to keep the event data flat, not nested. Nesting complicates querying and may impose other limitations. Flat data can be easily exported and analysed using other tools, which might be handy because analytics tools have limitations.

- It's usually better to have less event names, but it's not a hard rule. For example, we decided to have one `navigation` event that covers actions like opening modals. We could've went another route and defined multiple event names for different types of navigation (such as modals) but that would have created unnecessary complexity.

- If possible, it's preferable to aggregate only in the analytics tool, not in the code. That means the categorization of events should be preferably left to the analytics tool @mohsen-ghafouri. This will give us more flexibility, which is quite important now because we don't know how we want to consume the data or the business needs of the future. It's easy to group in analytics tools, but sometimes ~impossible~ _really hard_ to ungroup.

- Users of the analytics tool like Volo should have a word on the preferred approach on events and properties. Another reason for us to build a tracking plan and use that as a reference for implementation, not the other way around.

cc @seanstrom

Reply:
Hey @ilmotta, thanks for your feedback! This is exactly what I was saying, it’s more helpful to focus on the desired outcome. 
If we can clarify what the ideal outcome event should be, I can better determine where to trigger it. That’s why I created this doc https://www.notion.so/Event-Tracking-Specification-Mobile-11a8f96fb65c8032a05efcebea3a1ce6.

I already spend lots of time on metrics, and after all i can see we still have different option on basic info.

Reply:
@mohsen-ghafouri, that doc is a nice initiative 👍🏼 I hope we can grow it then as our mobile tracking plan. The specifics about analytics for dApps I think I will pass to you guys to decide. Above, I tried to outline a few ideas to take in consideration, especially in terms of keeping things simple and tracking less for now, which I think could be improved for this PR. We can always add, but removing or changing isn't convenient or possible sometimes.

To give a practical example, we could track very simply only the action of signing as approved or rejected, but not track if they were successful or not. We could also not track the method (e.g. `eth_sign`), that seems unnecessary at this stage too, but I'm sure there are reasons (it's not wrong), although we can always find a reason to track anything. In the `dapp-session` event, we might skip sending the networks because it seems unnecessary now.

Reply:
Let's have a meeting for tomorrow, sending an invite.